Modules/Filtering/BiasCorrection/test/itkN4BiasFieldCorrectionImageFilterTest.cxx

```diff
@@ -164,15 +164,44 @@ N4(int argc, char * argv[])
     maskImage->DisconnectPipeline();
   }
 
-  // instantiate N4 and assign variables not exposed to the user in this test.
+  // Instantiate N4 and assign variables not exposed to the user in this test.
   using CorrecterType = itk::N4BiasFieldCorrectionImageFilter<ImageType, MaskImageType, ImageType>;
   typename CorrecterType::Pointer correcter = CorrecterType::New();
-  correcter->SetSplineOrder(3);
+
+  EXERCISE_BASIC_OBJECT_METHODS(correcter, N4BiasFieldCorrectionImageFilter, ImageToImageFilter);
```

Comment:
CDash is failing here:
https://open.cdash.org/viewBuildError.php?buildid=7035321

The `Superclass` is [correctly declared](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Modules/Filtering/BiasCorrection/include/itkN4BiasFieldCorrectionImageFilter.h#L103).

Not sure if the class is missing the `ITK_TEMPLATE_EXPORT` macro.

Reply:
Still being reported [here](https://open.cdash.org/viewBuildError.php?buildid=7036964) after [this comment](https://github.com/InsightSoftwareConsortium/ITK/pull/2286#discussion_r574590519) was addressed. All three related tests passing locally on my Windows/VS setup.

Reply:
On [line 66](https://github.com/InsightSoftwareConsortium/ITK/blob/v5.2rc02/Modules/Core/TestKernel/include/itkTestingMacros.h#L66) it is `SuperClass` and on [line 69](https://github.com/InsightSoftwareConsortium/ITK/blob/v5.2rc02/Modules/Core/TestKernel/include/itkTestingMacros.h#L69) it is called `Superclass` (lowercase `c`).

But why does it occur only for a couple of classes?

Reply:
The one on line 66 is just the variable name, so although confusing (I would propose a separate `STYLE` improvement to avoid it in a separate PR), I think the issue is not there.

Reply:
Latest commit fixes this, but exposes some other issues (e.g. protected member access from test).

Reply:
OK, I see. Thanks for catching this ! Now, as you say, it is confusing that this has worked so far for the rest of the classes. I think it deserves some investigation. I am reopening the conversation until we are able to find a convincing explanation.  Thanks for your patience @dzenanz .

Reply:
Update: so the error is still present despite 107d1cc.

Reading `ITK_EXERCISE_BASIC_OBJECT_METHODS` again with more care, I'd revert 107d1cc: the version in `master` is correct: the method's `Class` and `SuperClass` arguments are used in [here](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Modules/Core/TestKernel/include/itkTestingMacros.h#L70) and [here](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Modules/Core/TestKernel/include/itkTestingMacros.h#L79) as strings to compare to the ones the object retrieves. I grant it might be clearer to rename the arguments to `ClassName` and `SuperclassName` in a separate PR to avoid confusion with capitals.

But the call `object->Superclass` is actually calling the instance of the object's parent class to retrieve the latter's `NameOfClass`, which is legal.

So there is something else with `itk::N4BiasFieldCorrectionImageFilter`/the issue is not in the definition of `ITK_EXERCISE_BASIC_OBJECT_METHODS`.